File: 160706/array/java/eight.java
// 8. 实现一个方法，传入字符串数组(偶数个)，返回a[0] => a[1]以此类推转换后的Hash
import java.util.HashMap;
public class eight{
  public static void main(String[] args) {
    String[] strAry = new String[]{"abc","bcd","cde","def"};
    HashMap<String,String> map = new HashMap<String, String>();
    map = exchange(strAry);
    System.out.println(map);
  }

  public static HashMap<String, String> exchange(String[] strAry){
    HashMap<String, String> map = new HashMap<String, String>();
    for (int i = 0; i < strAry.length; i ++) {
      if (i < strAry.length - 1) {
        map.put(strAry[i], strAry[i + 1]);
      }
    }
    return map;
  }
}

File: 160706/array/ruby/eight.rb
# 8. 实现一个方法，传入字符串数组(偶数个)，返回a[0] => a[1]以此类推转换后的Hash
str_ary = ["abc","bcd","cde","def","efg","efh"]
def coversion_ary_to_hash(str_ary)
  hash_sort_ary = {}
  str_ary.each do |ary|
    ary_index = str_ary.index(ary)
    if ary_index < str_ary.length - 1
      hash_sort_ary[ary] = str_ary[ary_index + 1]
    end
  end
  hash_sort_ary
end

converted_hash = coversion_ary_to_hash(str_ary)
p converted_hash